drivers/hdlc_rcp_if/Kconfig

```diff
@@ -0,0 +1,23 @@
+# HDLC RCP communication Interface configuration options
+
+# Copyright (c) 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# Openthead HDLC RCP communication Interface options
+#
+menuconfig HDLC_RCP_IF
+	bool "HDLC RCP communication Interface"
+	depends on NETWORKING
+	default y if NET_L2_PHY_IEEE802154
```

Comment:
I don't think this should be enabled by default in case **any** 15.4 L2 is available.

Reply:
Yes, i will change the default case